src/Symfony/Component/DependencyInjection/Extension/AbstractExtension.php

```diff
@@ -22,7 +23,7 @@
  *
  * @author Yonel Ceruto <yonelceruto@gmail.com>
  */
-abstract class AbstractExtension extends Extension implements ConfigurableExtensionInterface, PrependExtensionInterface
+abstract class AbstractExtension extends Extension implements ConfigurableExtensionInterface, PrependExtensionInterface, CompilerPassProvidingExtensionInterface, CompilerPassInterface
```

Comment:
i assume extensions implementing CompilerPassInterface are already accounted for in symfony? they should not be ignored by this PR

but i'd prefer users letting implement CompilerPassInterface as needed rather than knowing they can override default process() from AbstractExtension

Reply:
Indeed, `ExtensionCompilerPass` is intended to support extensions implementing the `CompilerPassInterface`. However, this is a separate topic from the one suggested by the title of the PR 😕.

Reply:
Ah, I somehow forgot that I contributed that `ExtensionCompilerPass` feature years ago :smile: 

I'll revert this change.